backend/kernelCI_app/views/hardwareDetailsView.py

```diff
@@ -153,79 +156,221 @@ class HardwareDetails(View):
+        if self.filterBuildDurationMin is not None and (
+            toIntOrDefault(duration, 0) < self.filterBuildDurationMin
+        ):
+            return False
 
-        return filters_map
+        return True
 
-    def pass_build_filter(self, data, filters):
-        for currentFilter in filters:
-            field = currentFilter.get("field")
-            value = currentFilter.get("value")
-            include_null = any(null_string in value for null_string in NULL_STRINGS)
-            build_field = f"build__{field}"
+    def __boots_filters_pass(self, status, duration):
+        if len(self.filterBootStatus) > 0 and (status not in self.filterBootStatus):
+            return False
+        if (
+            self.filterBootDurationMax is not None or self.filterBootDurationMin is not None
+        ) and duration is None:
+            return False
+        if self.filterBootDurationMax is not None and (
+            toIntOrDefault(duration, 0) > self.filterBootDurationMax
+        ):
+            return False
+        if self.filterBootDurationMin is not None and (
+            toIntOrDefault(duration, 0) < self.filterBootDurationMin
+        ):
+            return False
 
-            if include_null and data[build_field] is None:
-                return True
+        return True
 
-            if (data[build_field] not in value):
-                return False
+    def __non_boots_filters_pass(self, status, duration):
+        if len(self.filterTestStatus) > 0 and (status not in self.filterTestStatus):
+            return False
+        if (
+            self.filterTestDurationMax is not None or self.filterTestDurationMin is not None
+        ) and duration is None:
+            return False
+        if self.filterTestDurationMax is not None and (
+            toIntOrDefault(duration, 0) > self.filterTestDurationMax
+        ):
+            return False
+        if self.filterTestDurationMin is not None and (
+            toIntOrDefault(duration, 0) < self.filterTestDurationMin
+        ):
+            return False
 
         return True
 
-    def pass_test_filters(self, data, filters):
-        for currentFilter in filters:
-            field = currentFilter.get("field")
-            value = currentFilter.get("value")
-
-            if field == "path":
-                return value[0] in data[field]
+    def record_in_filter(
+        self,
+        record: Dict,
+    ) -> bool:
+        record_arch_null_accept = (
+            not record['build__architecture']
+            and any(null_string in self.filterArchitecture for null_string in NULL_STRINGS)
+        )
+        record_compiler_null_accept = (
+            not record['build__compiler']
+            and any(null_string in self.filterTreeDetailsCompiler for null_string in NULL_STRINGS)
+        )
+        record_config_null_accept = (
+            not record['build__config_name']
+            and any(null_string in self.filterTreeDetailsConfigs for null_string in NULL_STRINGS)
```

Comment:
Is there a reason here to not just use `self.filterArchitecture in NULL_STRINGS` (the same applies to the other instances of `any(null_string in _ for null_string in NULL_STRINGS)`?

Reply:
`self.filterArchitecture` is a set of architecture to filter.
`self.filterTreeDetailsCompiler` and `filterTreeDetailsConfigs` too